meta-lmp-base/recipes-bsp/trusted-firmware-a/tf-a-tools-fio_2.6.bb

```diff
@@ -0,0 +1,32 @@
+SUMMARY = "Cert_create & Fiptool for fip generation for Trusted Firmware-A"
+LICENSE = "BSD-3-Clause"
+LIC_FILES_CHKSUM = "file://license.rst;md5=1dd070c98a281d18d9eefd938729b031"
+
+SRC_URI = "git://github.com/ARM-software/arm-trusted-firmware.git;protocol=https;nobranch=1"
+#SRCREV corresponds to v2.6
+SRCREV = "a1f02f4f3daae7e21ee58b4c93ec3e46b8f28d15"
+
+# Mandatory fix to allow feeding password through command line
+SRC_URI += "file://0099-tools-allow-to-use-a-root-key-password-from-command-.patch"
+
+DEPENDS:class-nativesdk = "nativesdk-openssl"
```

Comment:
The recipe on meta-st-stm32mp is broken as it build tf-a-tools-native but don't have any dependence for it

Reply:
reworking and now and the missing part is add support for the `OPENSSL_DIR` handling in the `fiptool`. I will do it patching the makefile.

```
--- a/recipes-bsp/trusted-firmware-a/tf-a-tools_2.6.bb
+++ b/recipes-bsp/trusted-firmware-a/tf-a-tools_2.6.bb
@@ -2,31 +2,29 @@ SUMMARY = "Cert_create & Fiptool for fip generation for Trusted Firmware-A"
 LICENSE = "BSD-3-Clause"
 LIC_FILES_CHKSUM = "file://license.rst;md5=1dd070c98a281d18d9eefd938729b031"
 
-SRC_URI = "git://github.com/ARM-software/arm-trusted-firmware.git;protocol=https;nobranch=1"
-#SRCREV corresponds to v2.6
+SRC_URI = "git://github.com/ARM-software/arm-trusted-firmware.git;protocol=https;branch=master"
 SRCREV = "a1f02f4f3daae7e21ee58b4c93ec3e46b8f28d15"
 
 # Mandatory fix to allow feeding password through command line
 SRC_URI += "file://0099-tools-allow-to-use-a-root-key-password-from-command-.patch"
 
-DEPENDS:class-nativesdk = "nativesdk-openssl"
+DEPENDS = "openssl"
 
 S = "${WORKDIR}/git"
 
-do_compile() {
-    oe_runmake certtool fiptool
-}
+EXTRA_OEMAKE += "V=1"
+EXTRA_OEMAKE += "certtool"
+#EXTRA_OEMAKE += "fiptool"
+EXTRA_OEMAKE += "OPENSSL_DIR='${STAGING_INCDIR}/..'"
 
 do_install() {
     install -d ${D}${bindir}
-    # cert_create
-    install -m 0755 ${B}/tools/cert_create/cert_create ${D}${bindir}/cert_create
-    # fiptool
-    install -m 0755 ${B}/tools/fiptool/fiptool ${D}${bindir}/fiptool
+    install -m 0755 \
+        ${B}/tools/cert_create/cert_create \
+        ${B}/tools/fiptool/fiptool \
+        ${D}${bindir}
 }
 
-FILES:${PN}:class-nativesdk = "${bindir}/cert_create ${bindir}/fiptool"
-
-RDEPENDS:${PN}:class-nativesdk += "nativesdk-libcrypto"
+RDEPENDS:${PN} += "libcrypto"
 
 BBCLASSEXTEND += "native nativesdk"
```

Reply:
The pending issue on my side is fixed upstream with https://github.com/ARM-software/arm-trusted-firmware/commit/dd14d0f63f7d049cbcbfd3f8128f05e9a351bbe5 so back-porting this and I finaly have the recipe fixed in  meta-lmp main-next.
I will send PR to meta-st-stm32mp and maybe try to update it to 2.7